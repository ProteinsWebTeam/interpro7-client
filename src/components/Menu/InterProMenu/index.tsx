import React, { PropsWithChildren } from 'react';

import MenuItem from 'components/Menu/MenuItem';
import { InterPro } from 'menuConfig';

import cssBinder from 'styles/cssBinder';

import fonts from 'EBI-Icon-fonts/fonts.css';
import local from './styles.css';

const css = cssBinder(fonts, local);

type Props = PropsWithChildren<{
  className?: string;
}>;

const InterProMenu = ({ className, children }: Props) => {
  return (
    <ul className={`${className || ''} ${css('interpro-menu')}`}>
      {children}
      {InterPro.map(
        ({ to, name, icon, iconClass = 'generic', activeClass, exact }) => (
          <li key={name}>
            <MenuItem to={to} activeClass={activeClass} exact={exact}>
              {icon !== 'H' && (
                <i
                  data-icon={icon}
                  className={css('icon', `icon-${iconClass}`)}
                />
              )}
              {icon === 'H' && <span />}
              {name}
            </MenuItem>
          </li>
        ),
      )}
    </ul>
  );
};

export default InterProMenu;
