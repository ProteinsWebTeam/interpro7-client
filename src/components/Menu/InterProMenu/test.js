import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import InterProMenu from '.';

const renderer = new ShallowRenderer();

describe('<InterProMenu />', () => {
  test('should render', () => {
    renderer.render(<InterProMenu />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
