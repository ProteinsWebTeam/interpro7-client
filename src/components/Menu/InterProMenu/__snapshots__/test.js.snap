// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InterProMenu /> should render 1`] = `
<ul
  className=" interpro-menu"
>
  <li>
    <Memo(Connect(MenuItem))
      exact={true}
      to={
        Object {
          "description": Object {},
        }
      }
    >
      <span />
      Home
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "main": Object {
              "key": "search",
            },
          },
        }
      }
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      Search
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      activeClass={[Function]}
      to={[Function]}
    >
      <i
        className="icon icon-common"
        data-icon="b"
      />
      Browse
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      to={[Function]}
    >
      <i
        className="icon icon-common"
        data-icon="*"
      />
      Results
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "other": Array [
              "release_notes",
            ],
          },
        }
      }
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      Release notes
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "other": Array [
              "download",
            ],
          },
        }
      }
    >
      <i
        className="icon icon-common"
        data-icon="="
      />
      Download
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "other": Array [
              "help",
            ],
          },
        }
      }
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      Help
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "other": Array [
              "about",
            ],
          },
        }
      }
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      About
    </Memo(Connect(MenuItem))>
  </li>
  <li>
    <Memo(Connect(MenuItem))
      to={
        Object {
          "description": Object {
            "other": Array [
              "settings",
            ],
          },
        }
      }
    >
      <i
        className="icon icon-common"
        data-icon=""
      />
      Settings
    </Memo(Connect(MenuItem))>
  </li>
</ul>
`;
